### PR TITLE
feat: dedicated ITS script

### PR DESCRIPTION
**Pull request by deanamiel**
Dedicated ITS script as per [AXE-2437](https://axelarnetwork.atlassian.net/jira/software/c/projects/AXE/boards/21?selectedIssue=AXE-2437)

[AXE-2437]: https://axelarnetwork.atlassian.net/browse/AXE-2437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: evm/cli-utils.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const addBaseOptions = (program, options = {}) => {
     }
 
     if (options.address) {
-        program.addOption(new Option('-a, --address <address>', 'override address'));
+        program.addOption(new Option('--address <address>', 'override address'));
```
Comment (milapsheth) [inline evm/cli-utils.js:27]: can we keep this and remove -a for artifact path?

File: evm/its.js [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+                printWarn('Event not emitted in receipt.');
+            }
+
+            break;
+        }
+
+        case 'callContractWithInterchainToken': {
+            const isPaused = await interchainTokenService.paused();
+
+            if (isPaused) {
```
Comment (milapsheth) [inline evm/its.js:439]: not a very important check. we'd know if we paused it

File: evm/its.js [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+                throw new Error(`${action} invalid while service is paused.`);
+            }
+
+            const { tokenId, destinationChain, destinationAddress, amount, data } = options;
+
+            if (!isValidTokenId(tokenId)) {
```
Comment (milapsheth) [inline evm/its.js:445]: Lot of code duplication. Some checks you can perform in the beginning if the option was provided, or create reusable functions

File: evm/its.js [hunk 4]
```diff
@@ -0,0 +1,723 @@
...
+
+            const tokenIdBytes32 = hexZeroPad(tokenId.startsWith('0x') ? tokenId : '0x' + tokenId, 32);
+
+            const tokenManagerAddress = await interchainTokenService.tokenManagerAddress(tokenIdBytes32);
+            printInfo(`TokenManager address for tokenId: ${tokenId}:`, tokenManagerAddress);
+
+            break;
+        }
+
+        case 'validTokenManagerAddress': {
```
Comment (milapsheth) [inline evm/its.js:96]: not needed, we can just print in the above method if token manager actually exists or not

File: evm/its.js [hunk 5]
```diff
@@ -0,0 +1,723 @@
...
+
+            const receipt = await tx.wait(chain.confirmations);
+
+            const eventEmitted =
+                wasEventEmitted(receipt, interchainTokenService, 'TokenManagerDeployed') ||
+                wasEventEmitted(receipt, interchainTokenService, 'TokenManagerDeploymentStarted');
+
+            if (!eventEmitted) {
+                printWarn('Event not emitted in receipt.');
+            }
```
Comment (milapsheth) [inline evm/its.js:245]: if it was a tx, then wait + event check can be done generically at the end for all cases, event to be checked can be defined here

File: evm/its.js [hunk 6]
```diff
@@ -0,0 +1,723 @@
...
+        }
+
+        case 'deployTokenManager': {
+            const isPaused = await interchainTokenService.paused();
+
+            if (isPaused) {
+                throw new Error(`${action} invalid while service is paused.`);
+            }
+
+            const { salt, destinationChain, type, params, gasValue } = options;
```
Comment (milapsheth) [inline evm/its.js:210]: Shall we just do arbitrary JSON encoded args instead?

File: evm/utils.js [hunk 7]
```diff
@@ -293,6 +293,35 @@ function isValidCalldata(input) {
     return hexPattern.test(input.slice(2));
 }
 
+/**
+ * Checks if a given string is a valid tokenId.
+ *
+ * @param {string} input - The input string to check.
+ * @returns {boolean} - True if the input is a valid tokenId, false otherwise.
+ */
+function isValidTokenId(input) {
```
Comment (milapsheth) [inline evm/utils.js:302]: this is not a general utils method. it should be in ITS script

**Review by milapsheth — CHANGES_REQUESTED** [review]

File: evm/its.js (hunk 5 above)
Comment (deanamiel) [reply evm/its.js:245]: moved this check to a separate function

File: evm/its.js (hunk 6 above)
Comment (deanamiel) [reply evm/its.js:186]: Mind elaborating here?

File: evm/its.js [hunk 8]
```diff
@@ -0,0 +1,479 @@
...
+
+async function handleTx(tx, chain, contract, action, firstEvent, secondEvent) {
+    printInfo(`${action} tx`, tx.hash);
+
+    const receipt = await tx.wait(chain.confirmations);
+
+    const eventEmitted =
+        (firstEvent ? wasEventEmitted(receipt, contract, 'TokenManagerDeployed') : true) ||
+        (secondEvent ? wasEventEmitted(receipt, contract, 'TokenManagerDeploymentStarted') : false);
+
```
Comment (milapsheth) [inline evm/its.js:49]: why are the event names hardcoded?

**Review by milapsheth — APPROVED** [review]

File: evm/its.js (hunk 8 above)
Comment (deanamiel) [reply evm/its.js]: good catch! will fix

File: .github/workflows/test.yaml [hunk 9]
```diff
@@ -65,12 +65,13 @@ jobs:
...
           }' > ./axelar-chains-config/info/local.json
 
-      # Create .env file with default hardhat private key that's prefunded
+      # Create .env file with default hardhat private key that's prefunded, default salt is keccak256('salt')
       - name: Prepare .env
         run: |
           echo 'PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80' >> .env
           echo 'ENV=local' >> .env
           echo 'CHAINS=test' >> .env
+          echo 'SALT=0xa05e334153147e75f3f416139b5109d1179cb56fef6a4ecb4c4cbc92a7c37b70' >> .env
```
Comment (milapsheth) [inline .github/workflows/test.yaml:74]: what's this for? Also, salt is usually a string and we hash in the script. If it's only needed by a specific script, then it should be passed in there and not applied globally to all cmds below

Comment (deanamiel) [reply .github/workflows/test.yaml]: will change to provide both `rawSalt` and `salt` options

**Pull request merged**